test/process_job_test.rb

```diff
@@ -53,6 +53,106 @@ def test_failed_setup
       end
     end
 
+    def test_failed_setup_retried_thrice
+      job = Jobs::TestRunnerJob.new(
+        SecureRandom.hex,
+        "ruby",
+        "bob",
+        { 'submission_filepaths' => [] },
+        "v1"
+      )
+
+      begin
+        ExecDocker.any_instance.stubs(docker_run_command: "#{__dir__}/bin/mock_docker")
+
+        ToolingInvoker::SetupInputFiles.expects(:call).
+          times(4).
```

Comment:
Use `MAX_NUM_RETRIES`?

Reply:
I'd prefer to have this hardcoded in the tests, as otherwise we could make a mistake updating the constant (e.g. setting it to 100) and we'd not have failing tests.